### ISSUE TITLE
`teleport configure --test` does not catch application name issues

### ISSUE THREAD

**Issue by webvictim** [gravitational/teleport#11894]
### Description

**What happened**: I configured an `app_service` in `/etc/teleport.yaml`:

```yaml
app_service:
  enabled: "yes"
  apps:
  - name: Github
    description: Github redirection
    uri: https://github.com
```

I then tried to start Teleport. I got an error: 

```console
Apr 12 12:47:44 ip-10-0-131-171 teleport[711779]: ERROR REPORT:
Apr 12 12:47:44 ip-10-0-131-171 teleport[711779]: Original Error: *trace.BadParameterError application name &#34;Github&#34; must be a valid DNS subdomain: https://goteleport.com/teleport/docs/application-access/#application-name
Apr 12 12:47:44 ip-10-0-131-171 teleport[711779]: Stack Trace:
Apr 12 12:47:44 ip-10-0-131-171 teleport[711779]:         /go/src/github.com/gravitational/teleport/lib/service/cfg.go:886 github.com/gravitational/teleport/lib/service.(*App).CheckAndSetDefaults
Apr 12 12:47:44 ip-10-0-131-171 teleport[711779]:         /go/src/github.com/gravitational/teleport/lib/config/configuration.go:1212 github.com/gravitational/teleport/lib/config.applyAppsConfig
Apr 12 12:47:44 ip-10-0-131-171 teleport[711779]:         /go/src/github.com/gravitational/teleport/lib/config/configuration.go:415 github.com/gravitational/teleport/lib/config.ApplyFileConfig
Apr 12 12:47:44 ip-10-0-131-171 teleport[711779]:         /go/src/github.com/gravitational/teleport/lib/config/configuration.go:1710 github.com/gravitational/teleport/lib/config.Configure
Apr 12 12:47:44 ip-10-0-131-171 teleport[711779]:         /go/src/github.com/gravitational/teleport/tool/teleport/common/teleport.go:325 github.com/gravitational/teleport/tool/teleport/common.Run
Apr 12 12:47:44 ip-10-0-131-171 teleport[711779]:         /go/src/github.com/gravitational/teleport/tool/teleport/main.go:34 main.main
Apr 12 12:47:44 ip-10-0-131-171 teleport[711779]:         /opt/go/src/runtime/proc.go:255 runtime.main
Apr 12 12:47:44 ip-10-0-131-171 teleport[711779]:         /opt/go/src/runtime/asm_amd64.s:1581 runtime.goexit
Apr 12 12:47:44 ip-10-0-131-171 teleport[711779]: User Message: application name &#34;Github&#34; must be a valid DNS subdomain: https://goteleport.com/teleport/docs/application-access/#application-name
```

However, testing the config with `teleport configure --test /etc/teleport.yaml` says that everything is OK:

```
$ sudo teleport configure --test /etc/teleport.yaml
OK /etc/teleport.yaml
```

Teleport refused to start at all until this error was fixed, which means that connection via all other Teleport services would be disrupted even though the config seems to validate.

**What you expected to happen**: If Teleport cannot start with the provided config, this should be caught by `teleport configure --test`. Otherwise, there is no way to be sure that a given config is valid after updates are applied until Teleport restarts, at which point it is too late.

### Reproduction Steps
See above.

### Server Details
- Teleport version (run `teleport version`): `Teleport v9.0.4 git:v9.0.4-0-gf577413d3 go1.17.7`
- Server OS (e.g. from `/etc/os-release`): `Ubuntu 21.10`
- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware): AWS EC2

**Comment by zmb3** [comment]
This is a more general problem.

`teleport configure --test` just attempts to load the config. It doesn't start any services, so it can't catch any errors that would occur at runtime.

It's basically only good for checking:

- is it valid YAML syntax with the appropriate keys
- are the ciphers, KEX/MAC algos, and CA signature algo valid

tl;dr - this test only checks for errors loading the config, not for errors _acting_ on the config.

**Comment by webvictim** [comment]
Yeah, I'm aware - just basically pointing out that the general expectation of a `--test` argument would be that if the test passes, Teleport will actually run with said config. Probably tricky to fix but definitely table stakes for any Enterprise-grade product offering reliable infrastructure access.

**Comment by zmb3** [comment]
Potential duplicate of #3248
